src/Controller/TrickController.php

```diff
@@ -38,9 +75,13 @@ public function show(Trick $trick)
      * @return \Symfony\Component\HttpFoundation\RedirectResponse|\Symfony\Component\HttpFoundation\Response
      * @throws \Exception
      */
-    public function createOrEdit(Request $request, ObjectManager $manager, Trick $trick = null)
+    public function editTrick(Request $request, ObjectManager $manager, Trick $trick = null)
     {
-        $trick = $trick ?? new Trick();
+        $user = $this->getUser();
+
+        if (!$user || !$this->isGranted(Member::ROLE_EDITOR) && !$user->isAuthor($trick)) {
```

Comment:
Use a Voter to hide the complexity of the access checks

https://symfony.com/doc/current/security/voters.html